networking/external_dns_operator/nw-control-dns-records-public-hosted-zone-azure.adoc

```diff
@@ -0,0 +1,135 @@
+openshift-console          downloads           downloads-openshift-console.apps.test-azure.qe.azure.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+----
+
+.  Get the dns zone list with respect to your resource group.
++
+[source,terminal]
+----
+$ az network dns zone list --resource-group "${RESOURCE_GROUP}"
+----
+
+. Create a dns zone for your `resourcegroup`.
++
+[source,terminal]
+----
+$ az network dns zone create -g "${RESOURCE_GROUP}" -n <ROUTE_ZONE_NAME>
+----
+
+. Create ExternalDNS CR.
++
+[source,yaml]
+----
+  apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-azure <1>
+spec:
+  domains:
+  - filterType: Include <2>
+    matchType: Exact <3>
+    name: test-azure1.qe.azure.devcluster.openshift.com <4>
+  provider:
+    type: Azure <5>
+  source:
+    openshiftRouteOptions: <6>
+      routerName: default <7>
+    fqdnTemplate:
+    - "{{.Name}}.apps.test-azure1.qe.azure.devcluster.openshift.com" <8>
+    type: OpenShiftRoute <9>
+  zones:
+  - "myzoneid" <10>
+"/subscriptions/53b4f551-f0fc-4bea-8cba-11111111111/resourceGroups/test-azure1-nxkxm-rg/providers/Microsoft.Network/dnszones/test-azure1.qe.azure.devcluster.openshift.com" <12>
+----
+
+<1> Name of external dns CR which will cause the operator to create a deployment and thereby the pods with this name.
+<2> Checks if the domain is included as a part of the route hostname.
+<3> Defines how to check if a domain is a part of the route hostname.
+<4> Defines the value which will be compared against the route hostname.
+<5> To publish records on public hosted zone on azure public DNS.
+<6> You can define an option for route source under this field.
+<7> If the source is openshift-route then you can pass the ingress controller name. Based on this name external-dns will select the respective router from the route status and map that routerCanonicalHostname to the route host while creating a CNAME record.
+<8> By default hostname annotation policy is Ignore so you need to define a fqdn template.
+A templated string that's used to generate DNS names from sources that don't define a hostname themselves, or to add a hostname suffix when paired with the fake source (optional). Accepts comma separated list for multiple global FQDN.
+<9> Defines source route i.e dns records for routes will be created on your cloud DNS.
+<10> Define the zone ID
+
+. Check the records for OCP created routes using the following command:
++
+[source,terminal]
+----
+$ az network dns record-set list -g "${RESOURCE_GROUP}"  -z <zone name you created in step 7 eg - test-azure.qe.azure.devcluster.openshift.com>  > record-list-azure.txt
+----
+
+.  Create a route.
++
+[source,terminal]
+----
+$ oc new-app --docker-image=openshift/hello-openshift -l app=hello-openshift
+$ oc expose service/hello-openshift -l app=hello-openshift
+----
```

Comment:
In AWS doc we just say that the records can be checked in the AWS web console. Why do we create a dedicated route to be checked with `az` client? Cannot we just say the same thing: check in the web console?